### PR TITLE
Add sdformat-mjcf package

### PR DESCRIPTION
**Pull request by traversaro**
Add recipe for sdformat-mjcf pure python package.

PyPI page: https://pypi.org/project/sdformat-mjcf/
GitHub page: https://github.com/gazebosim/gz-mujoco/tree/main/sdformat_mjcf

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge/help-python the PR is ready for review.

File: recipes/sdformat-mjcf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - wheel
+    - setuptools
+  run:
+    - python >=3.7
+    - gz-math7-python
+    - sdformat13-python
+    - dm-control
+
+test:
+  imports:
```
Comment (dopplershift) [inline recipes/sdformat-mjcf/meta.yaml:33]: If you want to keep the test files around, you can use [`source_files`](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-files):
```suggestion
test:
  source_files:
    - sdformat_mjcf/tests
  imports:
```

File: recipes/sdformat-mjcf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - pip check
+    - sdf2mjcf --help
+    - mjcf2sdf --help
+    # Tests are not installed, so we re-download the source code to run the tests
+    - git clone -b v{{ version }} https://github.com/gazebosim/gz-mujoco gz-mujoco-test-src
+    - pytest ./gz-mujoco-test-src/sdformat_mjcf/ -v
+  requires:
+    - pip
+    - pytest
+    - git
```
Comment (dopplershift) [inline recipes/sdformat-mjcf/meta.yaml:45]:
```suggestion
    - pytest ./gz-mujoco-test-src/sdformat_mjcf/ -v
  requires:
    - pip
    - pytest
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just some minor changes to make testing easier and this is good to go.

**Comment by traversaro** [comment]
Thanks @dopplershift !

**Comment by traversaro** [comment]
@dopplershift with an additional change https://github.com/conda-forge/staged-recipes/pull/21855/commits/b8047c14e2eff5b3d925ec1665f7877b18dedd72 the `source_files` feature is working fine.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**